src/model.rs

```diff
@@ -202,6 +218,14 @@ impl Default for Model {
     }
 }
 
+impl Drop for Model {
+    fn drop(&mut self) {
+        self.vars.clear();
+        self.conss.clear();
+        scip_call! { ffi::SCIPfree(&mut self.scip) };
```

Comment:
In other methods, you copy the `self.scip` pointer. Do you have a guarantee that you don't have any other references to it anywhere else when you get there ? 
A quick look seems to imply that one could drop the model but keep variables, or a solution, that still contain a pointer to the scip struct.

You should probably be more careful with just copying raw pointers around. At least, write a comment explaining why you think it's safe every time. At best, just never do it, make a single rust struct for each type of C pointer, and then manage that memory with traditional rust memory management.